includes/models/model.llms.lesson.php

```diff
@@ -720,190 +719,216 @@ public function has_content() {
-					array(
-						'key'     => '_llms_order',
-						'value'   => $next_position,
-						'compare' => '=',
-					),
-				),
-			);
-			$sections = get_posts( $args );
+		/**
+		 * Filter the WP_Query arguments used to locate a sibling lesson for the specified lesson.
+		 *
+		 * @since [version]
+		 *
+		 * @param array       $args      WP_Query arguments array.
+		 * @param string      $direction Navigation direction. Either "prev" or "next".
+		 * @param LLMS_Lesson $lesson    Current lesson object.
+		 */
+		$args = apply_filters( 'llms_lesson_get_sibling_lesson_query_args', $args, $direction, $this );
 
-			if ( $sections ) {
-				$newsection = new LLMS_Section( $sections[0]->ID );
-				$lessons    = $newsection->get_lessons( 'posts' );
-			}
-		}
-		// phpcs:enable WordPress.DB.SlowDBQuery.slow_db_query_meta_key
-		// phpcs:enable WordPress.DB.SlowDBQuery.slow_db_query_meta_query
-		// phpcs:enable WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
+		$lessons = get_posts( $args );
 
 		return empty( $lessons ) ? false : $lessons[0]->ID;
+
 	}
 
 	/**
-	 * Get previous lesson ID
+	 * Performs a query to retrieve sibling lessons from the lesson's adjacent section
 	 *
-	 * @since 1.0.0
-	 * @since 3.24.0 Unknown.
-	 * @since 4.4.0 Improve query so that unpublished siblings do not break expected results.
-	 *              Use strict comparisions where needed.
-	 *              Make sure to always return `false` if no previous lesson is found.
-	 *              @since [version] Use a numeric comparison for the previous position meta query.
+	 * This will retrieve either the first lesson from the course's next section or the last
+	 * lesson from the course's previous section.
+	 *
+	 * @since [version]
 	 *
-	 * @return false|int ID of the previous lesson, if any, `false` otherwise.
+	 * @param string $direction Direction of navigation. Accepts either "prev" or "next".
+	 * @return false|int WP_Post ID of the sibling lesson or `false` if one doesn't exist.
 	 */
-	public function get_previous_lesson() {
+	private function get_sibling_section_query( $direction ) {
 
-		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_key
-		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_query
-		// phpcs:disable WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
+		$sibling_lesson = false;
+		$curr_section   = $this->get_section();
 
-		$parent_section   = $this->get_parent_section();
-		$current_position = $this->get_order();
+		// Ensure we're not working with an orphan.
+		if ( $curr_section ) {
 
-		$previous_position = $current_position - 1;
+			$curr_position = $curr_section->get_order();
 
-		if ( 0 !== (int) $previous_position ) { // `get_order()` returns an int, but it's filterable so let's be pedantic and cast it.
+			if ( 'next' === $direction ) {
+				$sibling_position = $curr_position + 1;
+				$order            = 'ASC';
+			} elseif ( 'prev' === $direction ) {
+				$sibling_position = $curr_position - 1;
```

Comment:
If the section's `$curr_position === 1` (and looking for a prev lesson) we wouldn't need to run the query at all, right?